File: src/main/java/com/dftm/model/PendingTask.java
package com.dftm.model;

import java.time.LocalDateTime;
import java.util.Map;
import java.util.HashMap;

import org.springframework.data.annotation.Id;
import org.springframework.data.mongodb.core.mapping.Document;

import lombok.AllArgsConstructor;
import lombok.Builder;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@Builder
@NoArgsConstructor
@AllArgsConstructor
@Document(collection = "pendingTasks")
public class PendingTask {
    @Id
    private String id;
    
    private String name;
    private String email;
    private String phone;
    private String address;
    private String apartment;
    private String description;
    
    // Användarfält för tilldelning
    private String assignedToUserId;
    private String assignedByUserId;
    
    // Översättningsfält
    private Language descriptionLanguage;
    @Builder.Default
    private Map<Language, String> descriptionTranslations = new HashMap<>();
    
    // Status kan vara "APPROVED" eller "REJECTED"
    private String status;
    
    // Datum då felanmälan mottogs
    private LocalDateTime received;
} 

File: frontend/src/types/pendingTask.ts
import { Language } from './language';

export interface PendingTask {
  id: string;
  name?: string;
  email?: string;
  phone?: string;
  address?: string;
  apartment?: string;
  title?: string;
  description: string;
  descriptionLanguage?: Language;
  descriptionTranslations?: Record<Language, string>;
  status: string;
  priority?: string;
  sender?: string;
  recipient?: string;
  messageId?: string;
  reporter?: string;
  createdAt?: string;
  updatedAt?: string;
  assigned?: boolean;
  approved?: boolean;
  received?: string;
} 